File: realm-common/src/main/java/io/github/solusmods/eternalcore/realm/impl/network/RealmNetwork.java
package io.github.solusmods.eternalcore.realm.impl.network;


import io.github.solusmods.eternalcore.network.api.util.NetworkUtils;
import io.github.solusmods.eternalcore.realm.impl.network.c2s.RequestRealmBreakthroughPacket;
import io.github.solusmods.eternalcore.realm.impl.network.c2s.SyncRealmStoragePayload;

public class RealmNetwork {

    private RealmNetwork() {
    }

    public static void init() {
        NetworkUtils.registerC2SPayload(RequestRealmBreakthroughPacket.TYPE,
                RequestRealmBreakthroughPacket.STREAM_CODEC, RequestRealmBreakthroughPacket::handle);
        NetworkUtils.registerC2SPayload(SyncRealmStoragePayload.TYPE, SyncRealmStoragePayload.STREAM_CODEC, SyncRealmStoragePayload::handle);
    }


}

File: realm-common/src/main/kotlin/io/github/solusmods/eternalcore/realm/impl/network/RealmNetwork.kt
package io.github.solusmods.eternalcore.realm.impl.network

import dev.architectury.networking.NetworkManager
import io.github.solusmods.eternalcore.network.api.util.NetworkUtils.registerC2SPayload
import io.github.solusmods.eternalcore.realm.impl.network.c2s.RequestRealmBreakthroughPacket
import io.github.solusmods.eternalcore.realm.impl.network.c2s.SyncRealmStoragePayload


object RealmNetwork {
    fun init() {
        registerC2SPayload<RequestRealmBreakthroughPacket?>(
            RequestRealmBreakthroughPacket.Companion.TYPE,
            RequestRealmBreakthroughPacket.Companion.STREAM_CODEC
        ) { obj: RequestRealmBreakthroughPacket?, context: NetworkManager.PacketContext? ->
            obj!!.handle(context!!)
        }
        registerC2SPayload<SyncRealmStoragePayload?>(
            SyncRealmStoragePayload.Companion.TYPE,
            SyncRealmStoragePayload.Companion.STREAM_CODEC
        ) { obj: SyncRealmStoragePayload?, context: NetworkManager.PacketContext? ->
            obj!!.handle(context!!)
        }
    }
}
